http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionCreator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.urlconnection;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+
+/**
+ * An interface that, given a {@link URI} creates a new {@link HttpURLConnection}. This allows customization
+ * of the creation and configuration of the {@link HttpURLConnection}.
+ */
+@FunctionalInterface
```

Comment:
Checkstyle would fail for missing sdk annotation

Reply:
Oh our checkstyle does not check for interface apparently. :\ Will fix it.
https://github.com/aws/aws-sdk-java-v2/blob/5005b0f940166e8099feb17b06ed85158a1dab0e/build-tools/src/main/java/software/amazon/awssdk/buildtools/checkstyle/MissingSdkAnnotationCheck.java#L45-L47